From c0726642f03c52051a1aeb5e90f59062011463f0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fernando=20Tarl=C3=A1=20Cardoso=20Lemos?=
 <fernandotcl@gmail.com>
Date: Fri, 11 May 2012 23:17:35 -0300
Subject: [PATCH] Output something, don't crash when input garbage.

If we receive a JSON map that doesn't have the full_text key, we now
consider full_text to be "ERROR". That is indicative that something
is wrong and needs to be fixed in i3status (or whatever is being used
to retrieve the status line text).

Fixes: #705
---
 i3bar/src/child.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/i3bar/src/child.c b/i3bar/src/child.c
index 0b6f07d..56ebec9 100644
--- a/i3bar/src/child.c
+++ b/i3bar/src/child.c
@@ -128,6 +128,8 @@ static int stdin_end_map(void *context) {
     parser_ctx *ctx = context;
     struct status_block *new_block = smalloc(sizeof(struct status_block));
     memcpy(new_block, &(ctx->block), sizeof(struct status_block));
+    if (!new_block->full_text)
+        new_block->full_text = strdup("ERROR");
     TAILQ_INSERT_TAIL(&statusline_head, new_block, blocks);
     return 1;
 }
-- 
1.7.10

